### PR TITLE
katana (and other swords) will now try to steathe when using them inhands

### PR DESCRIPTION
**Pull request by ZWRh3**
## About the PR 
swords will now try to steathe when using them inhands

try means loop over all valid steathes in your inventory
backpack contents and similar excluded on purpose
chest item included on purpose (chest item steathe is mostly a self handicapping gimmick rather than a practical op thing)

also fixes steathes not working if they are used while not inside of a hand (chest item bug fix)
## Why's this needed? 
makes it easier to conceal your weapon

the latter bug fix was done just because it leads to a dumb funny thing
## Changelog 

```changelog
(u)ZWRh3
(+)Swords will now try to steathe themselves if you use them inhands.
```

### CODE REVIEW

File: code/obj/item/misc_weapons.dm [hunk 1]
```diff
@@ -1027,6 +1027,12 @@
 	is_syndicate = TRUE
 	var/delimb_prob = 1
 	custom_suicide = 1
+/obj/item/swords/attack_self(mob/user)
```
Comment (pali6) [inline code/obj/item/misc_weapons.dm:1030]:
```suggestion

/obj/item/swords/attack_self(mob/user)
```

File: code/obj/item/misc_weapons.dm [hunk 2]
```diff
@@ -1028,6 +1028,13 @@
 	var/delimb_prob = 1
 	custom_suicide = 1
 
+/obj/item/swords/attack_self(mob/user)
+	. = ..()
+	for(var/obj/item/swords_sheaths/steath in user) //this working with chest items is intented according to my balance tests its ok its also funny
+		if(!steath.sword_inside && istype(src, steath.sword_path))
+			steath.Attackby(src,user)
```
Comment (pali6) [inline code/obj/item/misc_weapons.dm:1035]:
```suggestion
	for(var/obj/item/swords_sheaths/sheath in user) //this working with chest items is intented according to my balance tests its ok its also funny
		if(!sheath.sword_inside && istype(src, sheath.sword_path))
			sheath.Attackby(src, user)
```

**Review by pali6 — APPROVED** [review]

**Comment by pali6** [comment]
@Tarmunora do you see any potential balance issues with this (either in general or the chest-sheath thing)?

**Comment by ZWRh3** [comment]
tarm if you are reading this: chest steathing is a self handicap as any staffie can just force you to take the sword out of your ass (even if stunned) making it impossible to properly hide it that part is just a dumb gimmick

**Comment by Tarmunora** [comment]
> @Tarmunora do you see any potential balance issues with this (either in general or the chest-sheath thing)?

Indifferent overall but the chest sheath thing is really dumb and shouldn't be added imo.

**Comment by ZWRh3** [comment]
> > @Tarmunora do you see any potential balance issues with this (either in general or the chest-sheath thing)?
> 
> Indifferent overall but the chest sheath thing is really dumb and shouldn't be added imo.

its a byproduct of a bug fix but it could be explicitly disabled

**Pull request closed without merging**